components/renku_data_services/secrets_storage/secret/api.spec.yaml

```diff
@@ -0,0 +1,230 @@
+  version: 0.1.0
+servers:
+  - url: /api/data
+  - url: /ui-server/api/data
+paths:
+  /secrets/{user_id}:
+    get:
+      summary: Get all user's secrets
+      parameters:
+        - in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: List of all secrets
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/SecretsList"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - secrets_storage_api
+    post:
+      summary: Add a new secret
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SecretPost"
+      responses:
+        "201":
+          description: The secret was added
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Secret"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - secrets_storage_api
+  /secrets/{secret_id}:
+    get:
+      summary: Get a secret
+      parameters:
+        - in: path
+          name: secret_id
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: The secret
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Secret"
+        "404":
+          description: The user or secret doesn't exist
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - secrets_storage_api
+    patch:
+      summary: Update a secret's value and it's modification date
+      parameters:
+        - in: path
+          name: secret_id
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/SecretPatch"
+      responses:
+        "200":
+          description: The patched secret
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Secret"
+        "404":
+          description: The user or secret doesn't exist
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+        default:
+          $ref: "#/components/responses/Error"
+      tags:
+        - secrets_storage_api
+    delete:
+      summary: Remove a secret
+      parameters:
+        - in: path
+          name: user_id
+          required: true
+          schema:
+            type: string
```

Comment:
```suggestion
```
This endpoint doesn't have the `user_id` parameter in the path. Actually, instead of removing the parameter definition, we might want to add it to the path since `/secrets/{user_id}/{secret_id}` might be safer -- or it might be needed if `secret_id` might be a common string